src/ensembl/src/services/analytics-service.ts

```diff
@@ -19,12 +20,14 @@ class AnalyticsTracking {
 
   // Track an event
   public trackEvent(action: any) {
+    const { ga } = action.meta as AnalyticsType;
+
     this.reactGA.event({
-      action: action.meta.ga.action ? action.meta.ga.action : action.type,
-      category: action.meta.ga.category,
-      label: action.meta.ga.label,
-      value: action.meta.ga.value,
-      nonInteraction: action.meta.ga.nonInteraction
+      action: ga.action ? ga.action : action.type,
+      category: ga.category,
+      label: ga.label,
+      nonInteraction: ga.nonInteraction,
+      value: ga.value
```

Comment:
So, if you added the transport option to the AnalyticsOptions type, does it mean we should use it here?

Reply:
Yes, I think we will need to update it to contain the `transport` property.

Reply:
I have updated it with cd5a0a1 commit.

Reply:
The `transport` option is now not available for the user and instead it is specified in the GA event object.

Reply:
Here's the relevant section in GA documentation: https://developers.google.com/analytics/devguides/collection/analyticsjs/sending-hits#specifying_different_transport_mechanisms

I am not quite sure what they are using by default ("By default, analytics.js picks the HTTP method and transport mechanism with which to optimally send hits"), but suspect they are using image. If so, and if we want to keep it as image, there is no point in us even setting this option.

I guess there are only two reasons to have this field set:
- we want to always use XHR for reporting (good: we will see analytics requests in the xhr tab; bad: as said in the docs, xhr may not work if the user is unloading the page, but perhaps we don't care)
- we want to use beacon by default (plus: more reliable way of tracking; potential minus: we probably won't see analytics requests in xhr tab; also, I am unclear what GA will do if beacon is unsupported, as it is in IE, but I think it's safe to assume that GA will just default to a different available transport).

 Notice also that GA intends to eventually move to using beacons (which is the proper way of doing things anyway).

To sum up: either remove this field entirely or set it to xhr or beacon. No point in setting it to image.

Reply:
> To sum up: either remove this field entirely or set it to xhr or beacon. No point in setting it to image.

Yes, that's right. Are we okay to go with xhr for now?

Reply:
I left the transport property there so that we can change to whatever option we want later.

As for `xhr` being the default option, one downside I see with using it is that the network tab will have too many requests. This could make it a bit difficult when we want to debug XHR issues. Having said that, it looks better than the other two options.

Reply:
I have now set the default option to `xhr`